doc/api/assert.md

```diff
@@ -916,20 +920,25 @@ assert(0);
 <!-- YAML
 added: REPLACEME
 -->
-* `block` {Function}
-* `error` {RegExp|Function|Object}
+* `block` {Function|Promise}
+* `error` {RegExp|Function|Object|Error}
 * `message` {any}
 
-Awaits for promise returned by function `block` to be rejected.
+Awaits the `block` promise or, if `block` is a function, immediately call the
+function and await the returned promise to complete. It will then check that the
+promise is rejected.
 
-When `assert.rejects()` is called, it will immediately call the `block`
-function, and awaits for completion.
+If `block` is a function and it throws an error synchronously,
+`assert.rejects()` will return a rejected Promise with that error without
+checking the error handler.
 
 Besides the async nature to await the completion behaves identically to
 [`assert.throws()`][].
 
-If specified, `error` can be a constructor, [`RegExp`][], a validation
-function, or an object where each property will be tested for.
+If specified, `error` can be a [`Class`][], [`RegExp`][], a validation function,
+an object where each property will be tested for or an instance of error where
```

Comment:
nit: an object where each property will be tested for, or an instance
Ditto on line 1027